packages/flutter/lib/src/material/material.dart

```diff
@@ -684,8 +682,7 @@ abstract class InkFeature {
   ///
   /// Typically used by subclasses to call
   /// [MaterialInkController.markNeedsPaint] when they need to repaint.
-  MaterialInkController get controller => _controller;
-  final _RenderInkFeatures _controller;
+  final SplashController controller;
```

Comment:
The "private member/public getter" pattern is no longer necessary!

Reply:
I wonder what changed it? Was it forbidden to override `final` members?